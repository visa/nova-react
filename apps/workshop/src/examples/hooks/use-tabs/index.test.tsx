/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { axe } from 'jest-axe';

import metaData from './meta.json';

import { AutoSelectTabs } from './auto-select-tabs';
import { UseTabsExample } from './use-tabs-example';

const examples = [
  { Component: AutoSelectTabs, title: metaData['auto-select-tabs'].title },
  { Component: UseTabsExample, title: metaData['use-tabs-example'].title },
];

describe('useTabs example', () => {
  examples.forEach(({ Component, title }) => {
    it(`${title} should render correctly`, async () => {
      const { container } = render(<Component />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
      expect(container).toMatchSnapshot();
    });
  });

  describe('useTabs example', () => {
    it('should have active tab after click', () => {
      render(<UseTabsExample />);
      const tab = screen.getByText('Tab 2');
      fireEvent.click(tab);
      expect(tab).toHaveAttribute('aria-selected', 'true');
    });
    it('should have active tab after keyboard navigation', async () => {
      render(<UseTabsExample />);
      const user = userEvent.setup();
      const tab = screen.getByText('Tab 2');
      await user.tab();
      await user.keyboard('{ArrowRight}');
      await user.keyboard('{Enter}');
      expect(tab).toHaveAttribute('aria-selected', 'true');
    });
  });
  describe('auto-active horizontal tab', () => {
    it('should have active tab after click', () => {
      render(<AutoSelectTabs />);
      const tab = screen.getByText('Tab 2');
      fireEvent.click(tab);
      expect(tab).toHaveAttribute('aria-selected', 'true');
    });
    it('should have active tab after keyboard navigation', async () => {
      render(<AutoSelectTabs />);
      const user = userEvent.setup();
      const tab = screen.getByText('Tab 2');
      await user.tab();
      await user.keyboard('{ArrowRight}');
      expect(tab).toHaveAttribute('aria-selected', 'true');
    });
  });
});
