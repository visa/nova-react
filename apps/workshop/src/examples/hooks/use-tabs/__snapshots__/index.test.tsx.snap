// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useTabs example Auto-select tabs on arrow key press should render correctly 1`] = `
<div>
  <ul
    class="v-tabs v-tabs-horizontal"
    role="tablist"
  >
    <li
      class="v-tab"
      role="none"
    >
      <button
        aria-selected="true"
        class="v-button v-button-large v-button-tertiary"
        role="tab"
        tabindex="0"
      >
        Tab 1
      </button>
    </li>
    <li
      class="v-tab"
      role="none"
    >
      <button
        aria-selected="false"
        class="v-button v-button-large v-button-tertiary"
        role="tab"
        tabindex="-1"
      >
        Tab 2
      </button>
    </li>
    <li
      class="v-tab"
      role="none"
    >
      <button
        aria-selected="false"
        class="v-button v-button-large v-button-tertiary"
        role="tab"
        tabindex="-1"
      >
        Tab 3
      </button>
    </li>
    <li
      class="v-tab"
      role="none"
    >
      <button
        aria-selected="false"
        class="v-button v-button-large v-button-tertiary"
        role="tab"
        tabindex="-1"
      >
        Tab 4
      </button>
    </li>
  </ul>
</div>
`;

exports[`useTabs example Behavior and usage should render correctly 1`] = `
<div>
  <ul
    class="v-tabs v-tabs-horizontal"
    role="tablist"
  >
    <li
      class="v-tab"
      role="none"
    >
      <button
        aria-selected="false"
        class="v-button v-button-large v-button-tertiary"
        role="tab"
        tabindex="0"
      >
        Tab 1
      </button>
    </li>
    <li
      class="v-tab"
      role="none"
    >
      <button
        aria-selected="false"
        class="v-button v-button-large v-button-tertiary"
        role="tab"
        tabindex="-1"
      >
        Tab 2
      </button>
    </li>
    <li
      class="v-tab"
      role="none"
    >
      <button
        aria-selected="false"
        class="v-button v-button-large v-button-tertiary"
        role="tab"
        tabindex="-1"
      >
        Tab 3
      </button>
    </li>
  </ul>
</div>
`;
